Drop byteorder dependency

nabijaczleweli opened the pull request.

HeroicKatora commented:
That's good. For the sake of wording in release notes:
Did you observe noticable improvement in compile time?

nabijaczleweli commented:
I didn't measure beforehand because I just let the builds run in the background, but now that you mention it, yeah.

Purging `target/` and `Cargo.lock` every time, before:
```
nabijaczleweli@nabuter:/t/image-tiff$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

nabijaczleweli@nabuter:/t/image-tiff$ time cargo build
    Updating crates.io index
   Compiling byteorder v1.3.4
   Compiling adler32 v1.2.0
   Compiling lzw v0.10.0
   Compiling miniz_oxide v0.3.7
   Compiling tiff v0.6.0-alpha (T:\image-tiff)
    Finished dev [unoptimized + debuginfo] target(s) in 14.81s

real    0m15.291s
user    0m0.000s
sys     0m0.000s

nabijaczleweli@nabuter:/t/image-tiff$ time cargo build --release
    Updating crates.io index
   Compiling byteorder v1.3.4
   Compiling adler32 v1.2.0
   Compiling lzw v0.10.0
   Compiling miniz_oxide v0.3.7
   Compiling tiff v0.6.0-alpha (T:\image-tiff)
    Finished release [optimized] target(s) in 6.64s

real    0m6.695s
user    0m0.000s
sys     0m0.000s
```

After:
```
nabijaczleweli@nabuter:/t/image-tiff$ git checkout byteorder
Previous HEAD position was 418979f Merge pull request #86 from image-rs/release-0.6.0-alpha
Switched to branch 'byteorder'
Your branch is up to date with 'наб/byteorder'.

nabijaczleweli@nabuter:/t/image-tiff$ time cargo build
    Updating crates.io index
   Compiling adler32 v1.2.0
   Compiling lzw v0.10.0
   Compiling miniz_oxide v0.3.7
   Compiling tiff v0.6.0-alpha (T:\image-tiff)
    Finished dev [unoptimized + debuginfo] target(s) in 2.88s

real    0m2.926s
user    0m0.000s
sys     0m0.000s

nabijaczleweli@nabuter:/t/image-tiff$ time cargo build --release
    Updating crates.io index
   Compiling adler32 v1.2.0
   Compiling lzw v0.10.0
   Compiling miniz_oxide v0.3.7
   Compiling tiff v0.6.0-alpha (T:\image-tiff)
    Finished release [optimized] target(s) in 3.81s

real    0m3.863s
user    0m0.000s
sys     0m0.000s
```
